airbyte-integrations/connectors/source-presta-shop/source_presta_shop/streams.py

```diff
@@ -0,0 +1,679 @@
+
+
+class ShopUrls(PrestaShopStream):
+    """
+    Shop urls from multi-shop feature
+    https://devdocs.prestashop.com/1.7/webservice/resources/shop_urls/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "shop_urls"
+
+
+class Shops(PrestaShopStream):
+    """
+    Shops from multi-shop feature
+    https://devdocs.prestashop.com/1.7/webservice/resources/shops/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "shops"
+
+
+class SpecificPriceRules(PrestaShopStream):
+    """
+    Specific price rules management
+    https://devdocs.prestashop.com/1.7/webservice/resources/specific_price_rules/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "specific_price_rules"
+
+
+class SpecificPrices(PrestaShopStream):
+    """
+    Specific price management
+    https://devdocs.prestashop.com/1.7/webservice/resources/specific_prices/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "specific_prices"
+
+
+class States(PrestaShopStream):
+    """
+    The available states of countries
+    https://devdocs.prestashop.com/1.7/webservice/resources/states/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "states"
+
+
+class StockAvailables(PrestaShopStream):
+    """
+    Available quantities of products
+    https://devdocs.prestashop.com/1.7/webservice/resources/stock_availables/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "stock_availables"
+
+
+class StockMovementReasons(IncrementalPrestaShopStream):
+    """
+    The stock movement reason (Increase, Decrease, Custom Order, …)
+    https://devdocs.prestashop.com/1.7/webservice/resources/stock_movement_reasons/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "stock_movement_reasons"
+
+
+class StockMovements(IncrementalPrestaShopStream):
+    """
+    Stock movements management
+    https://devdocs.prestashop.com/1.7/webservice/resources/stock_movements/
+    """
+
+    data_key = "stock_mvts"
+    cursor_field = "date_add"
+
+    def path(self, **kwargs) -> str:
+        return "stock_movements"
+
+
+class Stores(IncrementalPrestaShopStream):
+    """
+    The stores
+    https://devdocs.prestashop.com/1.7/webservice/resources/stores/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "stores"
+
+
+class Suppliers(IncrementalPrestaShopStream):
+    """
+    The product suppliers
+    https://devdocs.prestashop.com/1.7/webservice/resources/suppliers/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "suppliers"
+
+
+class Tags(PrestaShopStream):
+    """
+    The Products tags
+    https://devdocs.prestashop.com/1.7/webservice/resources/tags/
+    """
+
+    def path(self, **kwargs) -> str:
+        return "tags"
+
+
+class TaxRuleGroups(IncrementalPrestaShopStream):
+    """
+    Group of Tax rule, along with their name
+    https://devdocs.prestashop.com/1.7/webservice/resources/tax_rule_groups/
+    """
+
+    def path(self, **kwargs) -> str:
```

Comment:
is it safe to say that here in many other places `self.path == self.name` ? maybe lets have default path implementation that return self.name?